cli/ocspserve/ocspserve.go

```diff
@@ -41,11 +45,47 @@ func ocspServerMain(args []string, c cli.Config) error {
 	}
 
 	log.Info("Registering OCSP responder handler")
+
 	http.Handle(c.Path, ocsp.NewResponder(src))
 
 	addr := fmt.Sprintf("%s:%d", c.Address, c.Port)
+	server := &http.Server{Addr: addr, Handler: nil}
+
 	log.Info("Now listening on ", addr)
-	return http.ListenAndServe(addr, nil)
+
+	//gracefull shutdown on SIGTERM or SIGINT
+	//see issue https://github.com/golang/go/issues/19541
+	//see https://play.golang.org/p/LdXUYyzDxY
+	exit := make(chan struct{})
+	quit := make(chan os.Signal)
+	signal.Notify(quit, syscall.SIGTERM)
+	signal.Notify(quit, syscall.SIGINT)
+	go func() {
+		defer func() {
+			if err := recover(); err != nil {
+				log.Infof("recovered: %+v\n", err)
+			}
+		}()
+		<-quit
+		d := time.Now().Add(5 * time.Second) // deadline 5s max
+		ctx, cancel := context.WithDeadline(context.Background(), d)
+
+		defer cancel()
+
+		log.Info("Shutting down server...")
+		if err := server.Shutdown(ctx); err != nil {
+			log.Fatalf("could not shutdown: %v", err)
+		}
+		close(exit)
+	}()
+	err = server.ListenAndServe()
+	<-exit
+	if err != http.ErrServerClosed {
+		log.Fatalf("listen: %s\n", err)
+		return err
+	} else {
```

Comment:
go vet complaint:
if block ends with a return statement, so drop this else and outdent its block

Reply:
I have changed in the new PR https://github.com/cloudflare/cfssl/pull/1272